src/wallet/rpcwallet.cpp

```diff
@@ -334,37 +334,32 @@ ReifyPrivacyPolicy(const std::optional<PrivacyPolicy>& defaultPolicy,
     return strategy;
 }
 
+bool
+InvolvesUnifiedAddress(
+        const std::optional<PaymentAddress>& sender,
+        const std::set<PaymentAddress>& recipients)
+{
+    bool hasUASender =
+        sender.has_value() && std::holds_alternative<UnifiedAddress>(sender.value());
+    bool hasUARecipient =
+        std::find_if(recipients.begin(), recipients.end(),
+                     [](const PaymentAddress& addr) {
+                         return std::holds_alternative<UnifiedAddress>(addr);
+                     })
+        != recipients.end();
+
+    return hasUASender || hasUARecipient;
+}
+
 // Determines which `TransactionStrategy` should be used for the “LegacyCompat”
 // policy given the set of addresses involved.
 PrivacyPolicy
-InterpretLegacyCompat(const ZTXOSelector& sender,
+InterpretLegacyCompat(const std::optional<PaymentAddress>& sender,
```

Comment:
It's important that this not be called until *after* we've done the check that the provided sender address does not correspond to a UA receiver for a UA in the wallet.

Reply:
I don’t think that’s true. We error in the case that the sender is a bare UA receiver (https://github.com/zcash/zcash/pull/6545/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dR4992). So we don’t have to worry about having, say, a taddr that results in a UA or account selector.

Reply:
It seems you added that condition in 1733c6bc0c5e26fcb260b322eddc67a275ddd94d, which is probably when the need for this cycle went away 🤞🏼.